syntax="proto3";

message Empty{}

message BlockLocationsReq {
  string path = 1;
}

message BlockLocationsResp {
  map <string, int32> block_entries = 1;  
  string error = 2;
}

message CalcAvgLoanReq {
  int32 county_code = 1;
}

message CalcAvgLoanResp {
  int32 avg_loan = 1;
  string source = 2; 
  string error = 3;
}

message StatusString{
  string status= 1;
}

service Lender {
  rpc DbToHdfs (Empty) returns (StatusString);
  rpc BlockLocations (BlockLocationsReq) returns (BlockLocationsResp);
  rpc CalcAvgLoan (CalcAvgLoanReq) returns (CalcAvgLoanResp);
}

